libraries/AP_Common/missing/byteswap.h

```diff
@@ -1,4 +1,4 @@
-#if defined(HAVE_BYTESWAP_H) && HAVE_BYTESWAP_H
+#if defined(HAVE_BYTESWAP_H) && HAVE_BYTESWAP_H && !defined(__ANDROID__)
```

Comment:
@lucasdemarchi i'm building with waf to generate the static library and then i link it with C++Builder toolchain from Embarcadero RAD Studio IDE. I have the macros declared on the ide, i'm finding the best way for development and make it easy too, but idon't know, i'm trying all options.
I think it's better declared externally and not internally on the IDE, i think.
The waf build system compile very good with Android NDK toolchain with this fixes.